src/cmds/qsub.c

```diff
@@ -5658,8 +5665,10 @@ do_daemon_stuff(void)
 		 * request could take a while to reach server and get processed
 		 * Qsub then does a regular submit (new connection)
 		 */
-		if ((time(0) - connect_time) > (CREDENTIAL_LIFETIME - QSUB_DMN_TIMEOUT))
-			goto out; /* connect credential likely timed out */
+		if (fl_unlinked == 0 && ((time(0) - connect_time) > (CREDENTIAL_LIFETIME - QSUB_DMN_TIMEOUT_LONG))) {
+	                unlink(fl);
+			fl_unlinked = 1;
```

Comment:
While fl_unlinked is technically correct, the real thing you are testing for is if the credential has timed out.  I think a better name for this variable is credential_timedout.